src/servers/src/influxdb.rs

```diff
@@ -141,14 +139,252 @@ impl TryFrom<&InfluxdbRequest> for InsertRequests {
+            let table_name = line.series.measurement.as_str();
+            let tags = &line.series.tag_set;
+            let fields = &line.field_set;
+            let ts = line.timestamp;
+            let num_columns = tags.as_ref().map(|x| x.len()).unwrap_or(0) + fields.len() + 1;
+
+            let TableData {
+                schema,
+                rows,
+                column_indexes,
+            } = table_data_map
+                .entry(table_name)
+                .or_insert_with(|| TableData {
+                    schema: Vec::with_capacity(num_columns),
+                    rows: Vec::new(),
+                    column_indexes: HashMap::with_capacity(num_columns),
+                });
+
+            let mut one_row = vec![Value { value_data: None }; schema.len()];
+
+            // tags
+            parse_tags(tags, column_indexes, schema, &mut one_row)?;
+            // fields
+            parse_fields(fields, column_indexes, schema, &mut one_row)?;
+            // timestamp
+            parse_ts(ts, value.precision, column_indexes, schema, &mut one_row)?;
+
+            rows.push(Row { values: one_row });
+        }
+
+        let inserts = table_data_map
+            .into_iter()
+            .map(
+                |(
+                    table_name,
+                    TableData {
+                        schema, mut rows, ..
+                    },
+                )| {
+                    let num_columns = schema.len();
+                    for row in rows.iter_mut() {
+                        if num_columns > row.values.len() {
+                            row.values.resize(num_columns, Value { value_data: None });
+                        }
+                    }
+
+                    RowInsertRequest {
+                        table_name: table_name.to_string(),
+                        rows: Some(Rows { schema, rows }),
+                        ..Default::default()
+                    }
+                },
+            )
+            .collect::<Vec<_>>();
+
+        Ok(RowInsertRequests { inserts })
+    }
+}
+
+fn parse_tags<'a>(
+    tags: &'a Option<TagSet>,
+    column_indexes: &mut HashMap<&'a str, usize>,
+    schema: &mut Vec<ColumnSchema>,
+    one_row: &mut Vec<Value>,
+) -> Result<(), Error> {
+    let Some(tags) = tags else {
+        return Ok(());
+    };
+
+    for (k, v) in tags {
+        let index = column_indexes.entry(k.as_str()).or_insert(schema.len());
+        if *index == schema.len() {
+            schema.push(ColumnSchema {
+                column_name: k.to_string(),
+                datatype: ColumnDataType::String as i32,
+                semantic_type: SemanticType::Tag as i32,
+            });
+            one_row.push(to_value(ValueData::StringValue(v.to_string())));
+        } else {
+            check_schema(ColumnDataType::String, SemanticType::Tag, &schema[*index])?;
+            // unwrap is safe here
+            one_row.get_mut(*index).unwrap().value_data =
```

Comment:
style(non-blocking): you could use `[]` which is the same as `get_mut().unwrap()`
```suggestion
            one_row[(*index].value_data =
```